sycl/source/detail/kernel_program_cache.cpp

```diff
@@ -0,0 +1,39 @@
+//==--- kernel_program_cache.cpp - Cache for kernel and program -*- C++-*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl/detail/kernel_program_cache.hpp>
+
+namespace cl {
+namespace sycl {
+namespace detail {
+kernel_program_cache::~kernel_program_cache() {
+  for (auto &ProgIt : m_CachedPrograms) {
+    ProgramWithBuildStateT &ProgWithState = ProgIt.second;
+    PiProgramT *ToBeDeleted = ProgWithState.Ptr.load();
+
+    if (!ToBeDeleted) {
```

Comment:
Minor.
You are using braces for one line body here but don't use on line 30. Please you please align this aspect.